vllm/core/scheduler.py

```diff
@@ -386,15 +402,13 @@ def _schedule_running(
         # groups to preempt.
         now = time.time()
         running_queue = policy.sort_by_priority(now, running_queue)
-
         while running_queue:
             seq_group = running_queue[0]
             num_running_tokens = self._get_num_new_tokens(
                 seq_group, SequenceStatus.RUNNING, enable_chunking, budget)
 
-            # We can have up to 1 running prefill at any given time in running
-            # queue, which means we can guarantee chunk size is at least 1.
-            assert num_running_tokens != 0
+            if num_running_tokens == 0:
```

Comment:
the assertion is true before because it assumes chunked prefill is always the last item in the queue (and there's only one). 

However, this is not true for preemption because chunked prefill could come to the beg of the queue if it is preempted by fcfs semantics.